File: src/main/java/com/mantenimiento/equipomedico/app/service/SolicitudServicioService.java
package com.mantenimiento.equipomedico.app.service;

import com.mantenimiento.equipomedico.app.entidad.SolicitudServicio;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface SolicitudServicioService {

    /**
     * Creación de una nueva solicitud de servicio.
     *
     * @param solicitudServicio
     * @return
     */
    SolicitudServicio create(SolicitudServicio solicitudServicio);

    /**
     * Edición de una solicitud de servicio existente.
     *
     * @param solicitudServicio
     * @return
     */
    SolicitudServicio update(SolicitudServicio solicitudServicio);

    /**
     * Obtiene la solicitud de servicio mediante su id.
     *
     * @param id
     * @return
     */
    SolicitudServicio get(Long id);

    /**
     * Obtiene todas las solicitudes de servicio.
     *
     * @return
     */
    List<SolicitudServicio> getAll();

}

File: client-app/src/app/service/solicitud-servicio.service.ts
import {Injectable} from '@angular/core';
import {ApiRequestService} from "./api-request.service";
import {SolicitudServicio} from "../domain/solicitud";
import {Observable} from "rxjs";
import {environment} from "../../environments/environment";

@Injectable()
export class SolicitudServicioService {

  private urlSolicitud = environment.service_uri + '/solicitudservicios';

  constructor(private apiRequest: ApiRequestService) {
  }

  /**
   * Se obtiene la lista de todas las solicitudes de servicios.
   * @returns {Observable<SolicitudServicio[]>}
   */
  getAllSolicitudServicios(): Observable<SolicitudServicio[]> {
    return this.apiRequest.get(this.urlSolicitud);
  }

  /**
   * Se obtiene la solicitud de servicio por su Id.
   * @param {number} solicitudId
   * @returns {Observable<SolicitudServicio>}
   */
  getSolicitudServicioById(solicitudId: number): Observable<SolicitudServicio> {
    const url = this.urlSolicitud + '/' + solicitudId;
    return this.apiRequest.get(url);
  }

  /**
   * Se crea una nueva solicitud de servicio.
   * @param servicio
   */
  crearSolicitudServicio(servicio: SolicitudServicio): Observable<SolicitudServicio> {
    const url = this.urlSolicitud;
    return this.apiRequest.post(url, servicio);
  }

  /**
   * Se actualizan los datos de una solicitud de servicio existente.
   * @param servicio
   */
  editarSolicitudServicio(servicio: SolicitudServicio): Observable<SolicitudServicio> {
    const url = this.urlSolicitud;
    return this.apiRequest.put(url, servicio);
  }


}
